Fix JaxRs Telemetry-1.1 EE7/8 Async

Currently, we have had to skip the async tests for JaxRsIntegrationWithConcurrency here for EE7 and EE8: 

https://github.com/OpenLiberty/open-liberty/blob/integration/dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/async/JaxRsServerAsyncTestServlet.java#L58-L66

We are getting the following error: 

```
###AssertionError Json Start###{"exceptionType":"AssertionError","message":"JaxRsServerAsyncTestServlet.testJaxRsServerAsyncCompletionStage: Expected 4 traces but found 3\nSpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=0e373407d673e84c, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=02ff0de2bbd61866, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name=\"jaxrsAsyncServer\", telemetry.sdk.language=\"java\", telemetry.sdk.name=\"opentelemetry\", telemetry.sdk.version=\"1.29.0\"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.openliberty.microprofile.telemetry, version=null, schemaUrl=null, attributes={}}, name=subtask, kind=INTERNAL, startEpochNanos=1698938670234524167, endEpochNanos=1698938673236658625, attributes=AttributesMap{data={test.baggage=test.value}, capacity=128, totalAddedValues=1}, totalAttributeCount=1, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}\nSpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=8221de4b3988c44a, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=2f5ad6e4914ea6a1, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name=\"jaxrsAsyncServer\", telemetry.sdk.language=\"java\", telemetry.sdk.name=\"opentelemetry\", telemetry.sdk.version=\"1.29.0\"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=Client filter, version=null, schemaUrl=null, attributes={}}, name=GET, kind=CLIENT, startEpochNanos=1698938670220604125, endEpochNanos=1698938673256577708, attributes=AttributesMap{data={net.peer.port=8010, http.response_content_length=2, http.url=http://localhost:8010/jaxrsAsyncServer/JaxRsServerAsyncTestEndpoint/completionstage, http.method=GET, http.status_code=200, net.peer.name=localhost, net.transport=ip_tcp}, capacity=128, totalAddedValues=7}, totalAttributeCount=7, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}\nSpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=2f5ad6e4914ea6a1, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=00000000000000000000000000000000, spanId=0000000000000000, traceFlags=00, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=false}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name=\"jaxrsAsyncServer\", telemetry.sdk.language=\"java\", telemetry.sdk.name=\"opentelemetry\", telemetry.sdk.version=\"1.29.0\"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.openliberty.microprofile.telemetry, version=null, schemaUrl=null, attributes={}}, name=testSpan-/jaxrsAsyncServer/testJaxrsServerAsync, kind=INTERNAL, startEpochNanos=1698938670209000000, endEpochNanos=1698938673257484792, attributes={}, totalAttributeCount=0, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}","stack":[{"className":"org.junit.Assert","methodName":"fail","fileName":"Assert.java","lineNumber":93},{"className":"io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter","methodName":"getFinishedSpanItems","fileName":"InMemorySpanExporter.java","lineNumber":94},{"className":"io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter$Proxy$_$$_WeldClientProxy","methodName":"getFinishedSpanItems","lineNumber":-1},{"className":"io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.async.JaxRsServerAsyncTestServlet","methodName":"doAsyncTest","fileName":"JaxRsServerAsyncTestServlet.java","lineNumber":90},{"className":"io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.async.JaxRsServerAsyncTestServlet","methodName":"testJaxRsServerAsyncCompletionStage","fileName":"JaxRsServerAsyncTestServlet.java","lineNumber":64},{"className":"jdk.internal.reflect.NativeMethodAccessorImpl","methodName":"invoke0","fileName":"NativeMethodAccessorImpl.java","lineNumber":-2},{"className":"jdk.internal.reflect.NativeMethodAccessorImpl","methodName":"invoke","fileName":"NativeMethodAccessorImpl.java","lineNumber":77},{"className":"jdk.internal.reflect.DelegatingMethodAccessorImpl","methodName":"invoke","fileName":"DelegatingMethodAccessorImpl.java","lineNumber":43},{"className":"java.lang.reflect.Method","methodName":"invoke","fileName":"Method.java","lineNumber":568},{"className":"componenttest.app.FATServlet","methodName":"doGet","fileName":"FATServlet.java","lineNumber":73}]}###AssertionError Json End###

[11/02/2023 15:24:45:359 GMT] 001 JaxRsIntegrationWithConcurrenc JaxRsServerAsyncTestServlet.te I <<< END   JaxRsServerAsyncTestServlet.testJaxRsServerAsyncCompletionStage_EE8_FEATURES_MicroProfile_41_MPTEL11
[11/02/2023 15:24:45:359 GMT] 001 FATRunner                      evaluate                       S null
java.lang.AssertionError: JaxRsServerAsyncTestServlet.testJaxRsServerAsyncCompletionStage: Expected 4 traces but found 3
SpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=0e373407d673e84c, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=02ff0de2bbd61866, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name="jaxrsAsyncServer", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.29.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.openliberty.microprofile.telemetry, version=null, schemaUrl=null, attributes={}}, name=subtask, kind=INTERNAL, startEpochNanos=1698938670234524167, endEpochNanos=1698938673236658625, attributes=AttributesMap{data={test.baggage=test.value}, capacity=128, totalAddedValues=1}, totalAttributeCount=1, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}
SpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=8221de4b3988c44a, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=2f5ad6e4914ea6a1, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name="jaxrsAsyncServer", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.29.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=Client filter, version=null, schemaUrl=null, attributes={}}, name=GET, kind=CLIENT, startEpochNanos=1698938670220604125, endEpochNanos=1698938673256577708, attributes=AttributesMap{data={net.peer.port=8010, http.response_content_length=2, http.url=http://localhost:8010/jaxrsAsyncServer/JaxRsServerAsyncTestEndpoint/completionstage, http.method=GET, http.status_code=200, net.peer.name=localhost, net.transport=ip_tcp}, capacity=128, totalAddedValues=7}, totalAttributeCount=7, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}
SpanData{spanContext=ImmutableSpanContext{traceId=e0f714ef4beb10e675b784d714d6487d, spanId=2f5ad6e4914ea6a1, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=00000000000000000000000000000000, spanId=0000000000000000, traceFlags=00, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=false}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={service.name="jaxrsAsyncServer", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.29.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.openliberty.microprofile.telemetry, version=null, schemaUrl=null, attributes={}}, name=testSpan-/jaxrsAsyncServer/testJaxrsServerAsync, kind=INTERNAL, startEpochNanos=1698938670209000000, endEpochNanos=1698938673257484792, attributes={}, totalAttributeCount=0, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}
	at org.junit.Assert.fail(Assert.java:93)
	at io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter.getFinishedSpanItems(InMemorySpanExporter.java:94)
	at io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter$Proxy$_$$_WeldClientProxy.getFinishedSpanItems(Unknown Source)
	at io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.async.JaxRsServerAsyncTestServlet.doAsyncTest(JaxRsServerAsyncTestServlet.java:90)
	at io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.async.JaxRsServerAsyncTestServlet.testJaxRsServerAsyncCompletionStage(JaxRsServerAsyncTestServlet.java:64)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:568)
	at componenttest.app.FATServlet.doGet(FATServlet.java:73)
[11/02/2023 15:24:45:359 GMT] 001 FATRunner                      evaluate                       I exiting io.openliberty.microprofile.telemetry.internal_fat.JaxRsIntegrationWithConcurrency.JaxRsServerAsyncTestServlet.testJaxRsServerAsyncCompletionStage```

--- a/dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/rest/TelemetryServletFilter.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/rest/TelemetryServletFilter.java
@@ -179,6 +179,8 @@ public class TelemetryServletFilter extends AbstractTelemetryServletFilter imple
 
                 @Override
                 public void onStartAsync(AsyncEvent event) throws IOException {
+                    // A new async cycle is starting, we need to re-register ourself
+                    event.getAsyncContext().addListener(this);
                 }
             });
         } else {

--- a/dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/async/JaxRsServerAsyncTestServlet.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/async/JaxRsServerAsyncTestServlet.java
@@ -58,15 +58,12 @@ public class JaxRsServerAsyncTestServlet extends FATServlet {
     private TestSpans testSpans;
 
     @Test
-    // ERROR with JaxRsIntegrationWithConcurrency: "Expected 4 traces but found 3"
-    @SkipForRepeat({TelemetryActions.MP14_MPTEL11_ID, TelemetryActions.MP41_MPTEL11_ID})
+    @SkipForRepeat(TelemetryActions.MP14_MPTEL11_ID)
     public void testJaxRsServerAsyncCompletionStage() {
         doAsyncTest(JaxRsServerAsyncTestEndpointClient::getCompletionStage);
     }
 
     @Test
-    // ERROR with JaxRsIntegrationWithConcurrency: "Expected 4 traces but found 3"
-    @SkipForRepeat({TelemetryActions.MP14_MPTEL11_ID, TelemetryActions.MP41_MPTEL11_ID})
     public void testJaxRsServerAsyncSuspend() {
         doAsyncTest(JaxRsServerAsyncTestEndpointClient::getSuspend);
     }

--- a/dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryServletFilter.java
+++ b/dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryServletFilter.java
@@ -175,6 +175,8 @@ public class TelemetryServletFilter extends AbstractTelemetryServletFilter imple
 
                 @Override
                 public void onStartAsync(AsyncEvent event) throws IOException {
+                    // A new async cycle is starting, we need to re-register ourself
+                    event.getAsyncContext().addListener(this);
                 }
             });
         } else {

